File: app/lib/src/models/device.dart
import 'package:app/src/models/measurement.dart';

class Device {
  String bleId;
  List<Measurement> measurements;
  String name;
  bool active;

  Device({
    this.bleId,
    this.measurements,
    this.name,
    this.active,
  });

  factory Device.fromJson(Map<String, dynamic> json) => Device(
        bleId: json["bleId"],
        measurements: List<Measurement>.from(
            json["measurements"].map((x) => Measurement.fromJson(x))),
        name: json["name"],
        active: json["active"],
      );

  List<Measurement> getLasBunchOfTime(double lastSeconds) {
    final filtered = measurements
        .where((x) =>
            DateTime.now().difference(x.datetime).inSeconds <= lastSeconds)
        .toList();
    return filtered;
  }
}

File: device_simulator/src/models/device.py
class Device:
    def __init__(self, name, ble_id, active=True, turned_on=False):
        self.name = name
        self.ble_id = ble_id
        self.measurements = []
        self.active = active
        self.turned_on = turned_on

    def to_dict(self):
        return {
            'name': self.name,
            'bleId': self.ble_id,
            'measurements': [measurement.to_dict() for measurement in self.measurements],
            'active': self.active,
            'turnedOn': self.turned_on
        }

    def add_measurement(self, measurement):
        self.measurements.append(measurement)
